zone/task_manager.cpp

```diff
@@ -630,117 +674,92 @@ void TaskManager::TaskQuestSetSelector(
 		SendTaskSelector(client, mob, task_list_index, task_list);
 	}
 	else {
-		// TODO: check color, I think this might be only for (Shared) Tasks, w/e -- think should be yellow
-		mob->SayString(
-			client,
-			Chat::Yellow,
-			MAX_ACTIVE_TASKS,
-			client->GetName()
-		);
+		client->MessageString(Chat::Yellow, NO_TASK_OFFERS, ".", ".", client->GetName());
 	}
 }
 
-// sends task selector to client
-void TaskManager::SendTaskSelector(Client *client, Mob *mob, int task_count, int *task_list)
+void TaskManager::SharedTaskSelector(Client *client, Mob *mob, int count, const int *tasks)
 {
-	if (client->ClientVersion() >= EQ::versions::ClientVersion::RoF) {
-		SendTaskSelectorNew(client, mob, task_count, task_list);
-		return;
-	}
-	// Titanium OpCode: 0x5e7c
-	LogTasks("TaskSelector for [{}] Tasks", task_count);
-	int player_level = client->GetLevel();
+	LogTasks("[UPDATE] SharedTaskSelector called for array size [{}]", count);
 
-	// Check if any of the tasks exist
-	for (int i = 0; i < task_count; i++) {
-		if (m_task_data[task_list[i]] != nullptr) {
-			break;
-		}
-	}
-
-	int valid_task_count = 0;
-
-	for (int i = 0; i < task_count; i++) {
-		if (!ValidateLevel(task_list[i], player_level)) {
-			continue;
-		}
-		if (client->IsTaskActive(task_list[i])) {
-			continue;
-		}
-		if (!IsTaskRepeatable(task_list[i]) && client->IsTaskCompleted(task_list[i])) {
-			continue;
-		}
-
-		valid_task_count++;
+	if (count <= 0 || client->HasTaskRequestCooldownTimer()) {
+		client->SendTaskRequestCooldownTimerMessage();
+		return;
 	}
 
-	if (valid_task_count == 0) {
+	// check if requester already has a shared task (no need to query group/raid members if so)
+	if (client->GetTaskState()->HasActiveSharedTask()) {
+		client->MessageString(Chat::Red, SharedTaskMessage::NO_REQUEST_BECAUSE_HAVE_ONE);
 		return;
 	}
 
-	SerializeBuffer buf(50 * valid_task_count);
+	// get group/raid member character data from db (need to query for character ids)
+	auto request = SharedTask::GetRequestCharacters(database, client->CharacterID());
```

Comment:
Future optimization note to add character ids to the group/raid apis so the query isn't necessary

Reply:
ack